src/metabase/api/dashboard.clj

```diff
@@ -592,35 +596,37 @@
-       ;; separate query for each Field (for parameters that are mapped to more than one Field)
-      (try
-        (let [results (map (if (seq query)
-                               #(chain-filter/chain-filter-search % constraints query :limit result-limit)
-                               #(chain-filter/chain-filter % constraints :limit result-limit))
-                           field-ids)
-              values (distinct (mapcat :values results))
-              has_more_values (boolean (some true? (map :has_more_values results)))]
-          ;; results can come back as [v ...] *or* as [[orig remapped] ...]. Sort by remapped value if that's the case
-          {:values          (if (sequential? (first values))
-                              (sort-by second values)
-                              (sort values))
-           :has_more_values has_more_values})
-        (catch clojure.lang.ExceptionInfo e
-          (if (= (:type (u/all-ex-data e)) qp.error-type/missing-required-permissions)
-            (api/throw-403 e)
-            (throw e))))))))
+     (binding [qp.perms/*card-id* (or qp.perms/*card-id*
+                                      (get-card-id-from-resolved-params resolved-params param-key))]
+       (let [constraints (chain-filter-constraints dashboard constraint-param-key->value)
+             field-ids   (param-key->field-ids dashboard param-key)]
+         (when (empty? field-ids)
+           (throw (ex-info (tru "Parameter {0} does not have any Fields associated with it" (pr-str param-key))
+                           {:param       (get (:resolved-params dashboard) param-key)
+                            :status-code 400})))
+         ;; TODO - we should combine these all into a single UNION ALL query against the data warehouse instead of doing a
+         ;; separate query for each Field (for parameters that are mapped to more than one Field)
+         (try
+           (let [results (map (if (seq query)
+                                #(chain-filter/chain-filter-search % constraints query :limit result-limit)
+                                #(chain-filter/chain-filter % constraints :limit result-limit))
+                              field-ids)
+                 values (distinct (mapcat :values results))
+                 has_more_values (boolean (some true? (map :has_more_values results)))]
+             ;; results can come back as [v ...] *or* as [[orig remapped] ...]. Sort by remapped value if that's the case
+             {:values          (if (sequential? (first values))
+                                 (sort-by second values)
+                                 (sort values))
+              :has_more_values has_more_values})
+           (catch clojure.lang.ExceptionInfo e
+             (if (= (:type (u/all-ex-data e)) qp.error-type/missing-required-permissions)
+               (api/throw-403 e)
+               (throw e)))))))))
 
```

Comment:
The main change is making sure that `qp.perms/*card-id*` is set to the card-id. Somehow it is missing when doing constrained chained filters.

This view of the diff is a lot easier to read:

<img width="1715" alt="Screen Shot 2022-08-16 at 3 00 28 PM" src="https://user-images.githubusercontent.com/343288/184984610-2fa67825-d0be-4f15-97d7-1bd2de093367.png">